TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/support/Observable.java

```diff
@@ -0,0 +1,69 @@
+package com.novoda.tpbot.support;
+
+import com.novoda.notils.exception.DeveloperError;
+
+import java.util.ArrayList;
+
+public abstract class Observable<T> {
+
+    private boolean hasChanged = false;
+    private final ArrayList<Observer<T>> observers;
+
+    protected Observable() {
+        observers = new ArrayList<>();
+    }
+
+    public synchronized Observable<T> attach(Observer<T> observer) {
+        if (observer == null) {
+            throw new DeveloperError("You cannot attach a null observer");
+        }
+
+        if (observers.contains(observer)) {
+            return this;
+        }
+
+        observers.add(observer);
+
+        return this;
+    }
+
+    public synchronized void detach(Observer observer) {
+        observers.remove(observer);
+    }
+
+    protected void notifyOf(T newValue) {
+        Observer<T>[] observersCopy;
+
+        synchronized (this) {
+            if (hasNotChanged()) {
+                return;
+            }
+
+            observersCopy = observers.toArray(new Observer[observers.size()]);
```

Comment:
Safety I guess. I didn't want someone to remove from the observers while I am trying to update. I guess that is guarded against because of the `synchronized` on the `detach` though so this won't happen. I'll remove 😄